tools/serve/serve.py

```diff
@@ -514,6 +514,12 @@ def start(self, init_func, host, port, paths, routes, bind_address, config, **kw
 
     def create_daemon(self, init_func, host, port, paths, routes, bind_address,
                       config, **kwargs):
+
+        global logger
```

Comment:
I think you need to call `set_logger(config.logger)`. Maybe we could just have an `ensure_logger(config)` function.

Reply:
done

Reply:
This looks like you forgot to update the patch?

Reply:
Indeed. I have pushed the correct patch now.